EXILED/Exiled.Events/Patches/Generic/FixStaminaStat.cs

```diff
@@ -0,0 +1,73 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="FixStaminaStat.cs" company="ExMod Team">
+// Copyright (c) ExMod Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.Events.Patches.Generic
+{
+    using Exiled.API.Features.CustomStats;
+    using HarmonyLib;
+    using Mirror;
+    using PlayerRoles.FirstPersonControl;
+    using UnityEngine;
+
+    /// <summary>
+    /// Fix for <see cref="CustomStaminaStat"/>.
+    /// </summary>
+    [HarmonyPatch(typeof(FpcStateProcessor), nameof(FpcStateProcessor.UpdateMovementState))]
+    internal static class FixStaminaStat
+    {
+#pragma warning disable SA1313
+        private static bool Prefix(FpcStateProcessor __instance, PlayerMovementState state, PlayerMovementState __result)
+#pragma warning restore SA1313
+        {
+            if (__instance._stat is not CustomStaminaStat stat)
+                return true;
+
+            float height = __instance._mod.CharacterControllerSettings.Height;
+            float num1 = height * __instance._mod.CrouchHeightRatio;
+
+            if (__instance.UpdateCrouching(state == PlayerMovementState.Crouching, num1, height) || __instance._firstUpdate)
+            {
+                __instance._firstUpdate = false;
+                float num2 = Mathf.Lerp(0.0f, (float)((height - num1) / 2.0), __instance.CrouchPercent);
+                float num3 = Mathf.Lerp(height, num1, __instance.CrouchPercent);
+                float radius = __instance._mod.CharController.radius;
+                __instance._mod.CharController.height = num3;
+                __instance._mod.CharController.center = Vector3.down * num2;
+                __instance._camPivot.localPosition = Vector3.up * (float)((num3 / 2.0) - num2 - radius + FpcStateProcessor.EyeHeight);
+            }
+
+            if (!NetworkServer.active || __instance._useRate == 0.0)
+            {
+                __result = state;
+                return false;
+            }
+
+            if (state == PlayerMovementState.Sprinting)
+            {
+                if (stat.CurValue > 0.0 && !__instance.SprintingDisabled)
+                {
+                    stat.CurValue = stat.Clamp(stat.CurValue - (Time.deltaTime * __instance.ServerUseRate));
+                    __instance._regenStopwatch.Restart();
+                    __result = PlayerMovementState.Sprinting;
+                    return false;
+                }
+
+                state = PlayerMovementState.Walking;
+            }
+
+            if (stat.CurValue >= stat.MaxValue)
+            {
+                __result = state;
+                return false;
+            }
+
+            stat.CurValue = stat.Clamp(stat.CurValue + (__instance.ServerRegenRate * Time.deltaTime));
+            __result = state;
+            return false;
+        }
+    }
+}
```

Comment:
I will review later this on why it's needed

Reply:
we need to make it not clamp from 0 to 1 (a transpiler would probably work better here)

Reply:
Idk about what is better now. We are changing game logic pretty much. We can use transpiler only to call method, if needed, that will set/return values to set. Changing all of these stuff to transpiler looks too complex to update